examples/vmwithcontracts/storage/storage.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package storage
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/state"
+
+	smath "github.com/ava-labs/avalanchego/utils/math"
+	mconsts "github.com/ava-labs/hypersdk/examples/vmwithcontracts/consts"
+)
+
+type ReadState func(context.Context, [][]byte) ([][]byte, []error)
+
+// State
+// / (height) => store in root
+//   -> [heightPrefix] => height
+// 0x0/ (balance)
+//   -> [owner] => balance
+// 0x1/ (hypersdk-height)
+// 0x2/ (hypersdk-timestamp)
+// 0x3/ (hypersdk-fee)
+
+const (
+	// Active state
+	balancePrefix   = 0x0
+	heightPrefix    = 0x1
+	timestampPrefix = 0x2
+	feePrefix       = 0x3
+)
```

Comment:
Could we put all of the prefixes/keys here and update the state layout comment to include programs storage?